File: java/Sixth/src/test/q6/CardCompanyTest.java
package test.q6;

public class CardCompanyTest {

    public static void main(String[] args) {

        CardCompany company = CardCompany.getInstance();

        Card myCard = company.createCard();
        Card yourCard = company.createCard();

        System.out.println(myCard + " 카드번호: " + myCard.getCardNum());
        System.out.println(yourCard + " 카드번호: " + yourCard.getCardNum());
    }
}

File: python/p_oop2/chap05/excercise/card/card_company_test.py
from card_company import CardCompany
from card import Card


def main():
    company = CardCompany.get_instance()  # 싱글톤 패턴

    my_card = company.create_card()    # 회사 메서드를 통한 card 생성
    your_card = company.create_card()  # 회사 메서드를 통한 card 생성

    print(my_card.get_card_number())    # 10001 출력
    print(your_card.get_card_number())  # 10002 출력


if __name__ == '__main__':
    main()
